### ISSUE TITLE
Implement and validate the frozen Context 1.3 candidate

### ISSUE THREAD

**Issue by sirgippy** [sirgippy/GippyRank4#115]
## Summary

Implement the frozen next-generation Context prior architecture selected by the transfer-continuity research series and validate that the production implementation reproduces the research candidate exactly.

The research phase is complete.

The selected candidate is:

> **Context 1.3 = existing non-RP Context + total returning production + incoming prior offensive usage + incoming prior DB defensive impact + DB availability**

The candidate is the P3 representation from the defensive position-group study.

This ticket should:

1. implement that feature contract in the production Context prior code path;
2. consume the production transfer-feature boundary introduced by #113 / PR #114;
3. remove the obsolete passing / receiving / rushing returning-production decomposition from the candidate;
4. reproduce the frozen research P3 evaluation;
5. verify all existing History fallback and inference invariants;
6. emit complete candidate model artifacts and documentation;
7. leave Context 1.3 ready for production activation beginning with the first target season that has a valid on-time preseason transfer snapshot.

This ticket must **not** perform any additional feature selection or tuning.

## Important production-timing constraint

Context 1.3 depends on transfer features whose production-safe source requires an immutable snapshot captured on or before August 15 of the target season.

The production snapshot pipeline was created after the August 15, 2026 cutoff.

Therefore:

> **Do not retroactively publish or regenerate the 2026 Context prior as Context 1.3 using a post-cutoff transfer snapshot.**

The existing 2026 Context 1.2 artifact remains the legitimate production artifact for that season.

This ticket implements and validates the 1.3 specification.

Actual production activation should occur for the first target season with a valid on-time transfer snapshot.

Do not change the public/current production methodology identifier from 1.2 merely because the candidate implementation exists.

## Frozen statistical architecture

Context 1.3 must preserve the statistical model used by the successful research candidate.

### History features

Unchanged:

- `lag2_z_mean`
- `lag3_z_mean`
- `long_run_z_mean`

### Context features

Use exactly:

#### Coaching

- `coach_tenure_seasons`

#### Recruiting

- `recruiting_class_rank`
- `recruiting_class_points`
- `recruiting_points_2y_mean`
- `recruiting_points_3y_mean`
- `recruiting_points_4y_mean`
- `recruiting_points_trend`

#### Team Talent

- `talent_composite`

#### Returning production

- `returning_pct_ppa`

#### Incoming offensive transfer experience

- `transfer_in_prior_usage_sum`

#### Incoming DB transfer impact

- `transfer_in_prior_defensive_impact_db_sum`
- `transfer_in_prior_defensive_impact_db_available`

## Features removed from Context 1.2

Do not include:

- `returning_pct_passing_ppa`
- `returning_pct_receiving_ppa`
- `returning_pct_rushing_ppa`

These were explicitly superseded by the research result favoring total returning production plus imported prior offensive usage.

Do not retain them for backward compatibility inside the 1.3 fit.

The old Context 1.2 implementation/artifacts should remain reproducible separately.

## Explicitly excluded transfer features

Do not include:

- outgoing prior transfer usage;
- net transfer usage;
- transfer counts;
- transfer ratings;
- transfer stars;
- transfer talent interactions;
- QB-specific transfer usage;
- DL / EDGE impact;
- LB impact;
- aggregate defensive impact;
- defensive experience;
- defensive availability outside the frozen DB availability feature.

The research branch is closed.

## Location versus scale

This is important.

The successful D5/P3 research model matched the production Context **location fit**:

### Location features

Use:

- all History features;
- all frozen Context 1.3 features.

### Scale features

Use:

- History features only.

Do not place Context 1.3 features into the scale model.

Specifically preserve the research fit contract equivalent to:

```python
location_feature_names = [
    *H_FEATURES,
    *CONTEXT_1_3_FEATURES,
]

scale_feature_names = [
    *H_FEATURES,
]
```

Do not re-run the old C 1.2 location/scale/both selection exercise.

The statistical placement is frozen.

## Other frozen model settings

Preserve:

- `DirectRankModel`;
- normal distribution family;
- penalty `0.25`;
- existing t-1 empirical quadrature;
- t-2 / t-3 history summaries;
- deterministic optimizer retry on iteration-limit failure;
- current training-only missing-value preprocessing;
- current missingness-indicator behavior;
- current H fallback for rank-history cold starts.

Do not retune the penalty or optimizer.

## Production transfer-feature input

Consume the canonical derived artifact from the #113 / PR #114 preprocessing boundary.

The Context model should not:

- fetch CFBD;
- parse portal records;
- join player identities;
- calculate DB impact;
- discover raw transfer files.

Those responsibilities belong exclusively to the preseason transfer-feature pipeline.

The Context preprocessing boundary should consume only:

- `transfer_in_prior_usage_sum`
- `transfer_in_prior_defensive_impact_db_sum`
- `transfer_in_prior_defensive_impact_db_available`

from that artifact.

Use the attach-only integration mechanism introduced by #114 where appropriate.

## Fail closed on missing production transfer features

For a target season intended to use Context 1.3:

- every FBS team-season must have a canonical transfer-feature row;
- the transfer artifact must come from a validated manifest;
- the target season must have an on-time production-safe snapshot;
- missing team-season feature rows must not silently become zeros;
- model fitting/inference must not fall back to a live endpoint.

If the target-season production transfer artifact is unavailable:

> Context 1.3 annual inference must refuse to run.

Do not silently fall back from Context 1.3 to 1.2 under the same model/version identifier.

## Historical training semantics

Historical 2021–2025 transfer research data cannot be made retrospectively cutoff-safe.

For candidate validation, continue using the frozen retrospective research representation already used by the D5/P3 studies.

Clearly distinguish:

### Historical research/training data

Retrospective reconstruction used to estimate and validate the candidate architecture.

### Target-season production inference data

Must come from the immutable on-time snapshot pipeline.

Do not describe historical transfer values as archived preseason snapshots.

Document the historical provenance caveat explicitly in model metadata.

## Candidate implementation

Prefer introducing a distinct Context 1.3 builder/specification rather than mutating the V1.2 script in place.

The existing:

`build_preseason_context_prior_v1_2.py`

must remain capable of reproducing the historical 1.2 artifacts.

A reasonable structure would be:

`build_preseason_context_prior_v1_3.py`

or an equivalent shared implementation with explicitly versioned specifications.

Avoid copy/paste where clean factoring is obvious, but do not refactor unrelated model infrastructure merely for aesthetics.

## Feature approval boundary

Update the Context feature approval mechanism so the 1.3 specification explicitly permits exactly the frozen 1.3 feature family.

Do not broaden a generic allowlist to every transfer feature produced during research.

The production model should fail if an unapproved research feature is requested.

## Provenance

Context 1.3 model metadata should identify the provenance class for each feature family.

At minimum document:

### Rank history

Unchanged existing provenance.

### Coaching

Unchanged existing provenance.

### Recruiting

Unchanged existing provenance.

### Team Talent

Unchanged existing provenance.

### Returning production

Use only total `returning_pct_ppa`.

### Incoming prior offensive usage

Source:

> production preseason transfer-feature artifact derived from frozen portal + prior usage snapshots.

Historical evaluation caveat:

> retrospective research reconstruction for seasons before the production snapshot process existed.

### Incoming DB defensive impact

Source:

> production preseason transfer-feature artifact derived from frozen portal, prior roster, and defensive game-player snapshots.

Definition:

> sum of prior DB defensive impact for incoming DB transfers, where player impact is the frozen equal-weight mean of within-season × DB-group standardized `log1p` tackles, passes defended, and interceptions.

### DB availability

Document the frozen semantics:

- no incoming DB transfers => `0` impact, availability `1`;
- all incoming DB transfers resolved => summed impact, availability `1`;
- one or more unresolved => neutral impact value, availability `0`.

## Backtest implementation parity

Before interpreting candidate performance, reproduce the exact frozen research P3 model through the production 1.3 statistical implementation.

Use:

- training through 2021;
- frozen scoring on 2022–2025;
- identical FBS target population;
- identical H fallback;
- identical penalty;
- identical Context-location / History-scale feature placement;
- frozen research transfer feature values.

Compare against the checked-in #112 P3 artifacts.

The candidate should reproduce, within strict numerical tolerance:

- aggregate metrics;
- annual metrics;
- target team-season population;
- fitted feature names;
- location/scale feature sets;
- standardized coefficient values;
- fallback population.

Use checked-in research artifacts as the canonical expected values rather than copying rounded numbers into tests.

Approximate expected aggregate NLL is approximately:

`4.5061`

but the artifact, not this rounded value, is authoritative.

## D5 parity

Also verify that removing the DB value + DB availability fields from the 1.3 candidate reproduces the frozen D5 representation from the decomposition study.

This guards against accidentally changing:

- total RP semantics;
- incoming usage semantics;
- base Context features;
- preprocessing;
- penalty;
- location/scale placement.

## Context 1.2 comparison

Produce a final implementation-level comparison of:

- Context 1.2;
- D5;
- Context 1.3 candidate / P3.

Use the same 2022–2025 team-season population wherever comparisons require it.

At minimum report:

- NLL;
- CRPS;
- expected-rank MAE;
- median-rank MAE;
- 80% interval coverage;
- 80% interval average width.

Report aggregate and annual values.

This is a verification/reporting exercise.

Do not use the results to redesign 1.3.

## Expected qualitative result

The implementation should recover the already-established research pattern:

- D5 materially improves modern Context performance relative to C 1.2;
- P3 provides a smaller additional improvement beyond D5;
- the DB improvement is persistent rather than concentrated solely in one frozen season.

If the production implementation does not reproduce that pattern within expected numerical tolerance:

> stop and treat the discrepancy as an implementation/parity bug.

Do not tune the model until it agrees with the research result.

## Rolling-origin regression check

The rolling-origin studies have already been performed and should not be repeated as a new selection exercise.

However, where inexpensive, run the frozen 1.3 implementation through the existing rolling harness and verify it reproduces the checked-in P3 rolling metrics from #112.

This is regression parity only.

Do not create new candidates.

## Prediction population and fallback

Preserve the current Context behavior:

- teams with sufficient rank history receive the Context model;
- rank-history cold starts use exact H fallback;
- Context 1.3 must not alter promotion/cold-start coverage;
- H predictions themselves must remain unchanged.

Add an assertion that the H fallback population is identical between C 1.2 and the 1.3 candidate for the historical evaluation panel.

## Annual inference interface

Implement the annual Context 1.3 inference path so it is ready for a future production target season.

The interface should require:

- target season;
- trained-through season = target season − 1;
- canonical preseason team-season feature table;
- canonical production transfer-feature artifact;
- validated cutoff/provenance metadata.

Target-season outcomes remain forbidden exactly as under C 1.2.

## 2026 guardrail

Add an explicit regression test or runtime guard demonstrating that the new implementation does not accidentally reinterpret the current 2026 artifact as Context 1.3.

Do not overwrite:

- existing Context 1.2 2026 annual predictions;
- existing 2026 preseason snapshot;
- existing 2026 weekly snapshots that derive from the 1.2 prior.

The 2026 published lineage must remain reproducible as C 1.2.

## Versioning

Implement the Context 1.3 specification explicitly, but preserve artifact compatibility.

Do not simply change:

```python
CONTEXT_PRIOR_VERSION = "1.2"
```

to `1.3` while making 1.2 artifacts invalid.

When production activation eventually occurs:

- 1.3 will become the current Context production version;
- 1.2 must remain in the supported historical artifact-version set.

For this candidate ticket, keep the distinction between:

- implemented/validated 1.3 specification;
- currently active production Context version.

If existing methodology infrastructure needs a candidate identifier, add one explicitly rather than lying about what is currently published.

## Candidate artifacts

Write candidate-specific artifacts to a clearly separate path, for example:

`data/processed/preseason/context_v1_3_candidate/`

or another explicit candidate namespace.

At minimum include:

- `model_spec.json`
- `model_report.json`
- `evaluation.json`
- `predictions.csv`
- `feature_provenance.json`
- `context_coverage.csv`
- fitted model metadata / coefficients
- parity report against D5/P3 research artifacts

Do not overwrite the current production Context 1.2 artifacts in this ticket.

## Model specification artifact

The candidate `model_spec.json` should clearly show the expected feature set:

```text
lag2_z_mean
lag3_z_mean
long_run_z_mean
coach_tenure_seasons
recruiting_class_rank
recruiting_class_points
recruiting_points_2y_mean
recruiting_points_3y_mean
recruiting_points_4y_mean
recruiting_points_trend
talent_composite
returning_pct_ppa
transfer_in_prior_usage_sum
transfer_in_prior_defensive_impact_db_sum
transfer_in_prior_defensive_impact_db_available
```

And explicitly record:

- Context features affect location only;
- History features affect both location and scale under the frozen research contract;
- penalty = `0.25`;
- distribution = normal;
- candidate spec version = `1.3`.

## Tests

Add focused tests for:

### Feature contract

- 1.3 contains exactly the frozen features;
- passing RP is absent;
- receiving RP is absent;
- rushing RP is absent;
- no rejected transfer research feature enters the model.

### Feature attachment

- transfer artifact attaches by exact team-season identity;
- duplicate rows fail;
- missing target rows fail;
- audit-only transfer columns cannot accidentally become model features.

### Statistical contract

- Context features are location-only;
- scale remains History-only;
- penalty remains `0.25`;
- deterministic optimizer retry remains unchanged.

### Research parity

- D5 implementation reproduces frozen D5;
- P3 implementation reproduces frozen #112 P3;
- annual and aggregate metrics remain within strict tolerance;
- coefficient names/order and standardized values are consistent.

### Population

- same 2022–2025 team-season keys;
- same H fallback keys;
- no target-season outcomes enter candidate features.

### Production cutoff

- future 1.3 inference requires an on-time transfer snapshot;
- late-only target snapshot is rejected;
- absent production transfer artifact is rejected.

### Versioning

- existing 1.2 artifacts remain accepted/rebuildable;
- candidate 1.3 metadata is distinguishable from active production metadata.

## Final report

Produce a concise candidate report answering:

1. Does the implementation reproduce frozen D5?
2. Does it reproduce frozen P3?
3. What exactly changed from Context 1.2?
4. What is the 2022–2025 metric delta versus C 1.2?
5. What incremental value does DB impact add beyond D5?
6. Are History fallback behavior and target populations unchanged?
7. Is the production inference path cutoff-safe?
8. What prevents Context 1.3 from being retroactively used for 2026?
9. Is the candidate ready for activation for the first legitimately snapshotted future season?

### Pass

Declare the Context 1.3 specification **validated and ready for future production activation** if:

- D5 parity passes;
- P3 parity passes;
- target populations match;
- H fallback is unchanged;
- statistical feature placement matches research;
- production inference consumes only the #114 artifact;
- target-season cutoff requirements fail closed;
- 1.2 historical artifacts remain reproducible.

### Fail

If any of those conditions fail:

- do not promote;
- do not tune around the discrepancy;
- report the implementation/provenance mismatch;
- fix parity first.

## Non-goals

Do not:

- search new features;
- rerun feature selection;
- retune penalty;
- retune location vs scale;
- revisit DB component weights;
- add other defensive positions;
- add outgoing transfer information;
- replace the History model;
- change cold-start behavior;
- alter the likelihood/posterior model;
- rewrite historical 2026 published artifacts;
- activate 1.3 using a post-cutoff 2026 transfer snapshot.

This ticket is about faithfully implementing the architecture already selected by research.

## Acceptance criteria

The ticket is complete when:

1. Context 1.3 exists as an explicit candidate specification.
2. Its feature set exactly matches frozen P3.
3. Its statistical placement exactly matches the research model.
4. It reproduces D5 and P3 historical research artifacts within strict tolerance.
5. It consumes the production-safe transfer-feature boundary without duplicating transfer-data logic.
6. It fails closed when a target-season production transfer snapshot is unavailable or late.
7. It leaves Context 1.2 and the 2026 publication lineage untouched and reproducible.
8. Candidate artifacts clearly document performance, provenance, coefficients, and versioning.
9. The final report states whether Context 1.3 is ready for production activation beginning with the first season captured by the on-time snapshot process.

If all criteria pass, feature/model research is finished.

The next work should be operational activation of Context 1.3 for the first eligible preseason, not another modeling experiment.
